release/models/system/openconfig-system-controlplane.yang

```diff
@@ -199,6 +214,186 @@ module openconfig-system-controlplane {
+                container state {
+                  description
+                    "control-plane protection remaining packet counters";
+                  config false;
+                  uses system-controlplane-built-in-limits-traffic-state;
+                }
+              }
+            }
+            container vendor {
+              description
+                "An anchor point for vendor and platform specific
+                extensions.
+                As each platform  and vendor has different implementation and internal
+                parts where packets may be dropped at any point in time. Providing
+                vendor-specific counters provides better visibility into control-plane
+                traffic.
+                The recommended usage of this container is to create an augment at
+                /system/control-plane-traffice/ingress/built-in-limits/vendor that contains additional
+                vendor/platform specific containers.
+                e.g.
+                augment /system/control-plane-traffice/ingress/built-in-limits/vendor {
+                  container <vendor name> {
+                    container <platform name> {
+                      container <traffic-group-name>
+                        container config {
+                          leaf packet-max-rate {
+                              ...
+                           }
+                          leaf packet-burst-size {
+                             ...
+                          }
+                          ...
+                          }
+                          container state {
+                            leaf packet-max-rate {
+                              ...
+                            }
+                            leaf packet-burst-size {
+                              ...
+                            }
+                            ...
+                            leaf aggregate-counter-name {
+                              type string;
+                              description
+                                \"One of abstract counters defined under ../abstaract-counters/,
+                                such 'bgp' or 'L2control'.
+                                This self-documentation allows client to assotiate multiple vendor
+                                counters with vendor-neutral, abstract coarse counter\";
+                            }
+                            uses system-controlplane-built-in-limits-traffic-state;
+                          }
+                        }
+                      }
+                    }
+                  }
+
+                In addition to above, more granular packet/octet counters, also information
+                about rate limit and allowd burst size for each counter should be exposed.
+                Furthermore ability to configure rate limit and allowd burst size for each
+                counter could be exposed.";
```

Comment:
I think we should assert the above structure is required and  the values for the vendor, platform and traffic group are up to the implementation to choose.

Reply:
Added some requirment text, mandating x-reference to aggregate counters and presence of drop/accept counters.